endpoints/cookie_sync.go

```diff
@@ -455,6 +467,20 @@ func (c *cookieSyncEndpoint) handleResponse(w http.ResponseWriter, tf usersync.S
 	})
 
 	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+	if setCookieDeprecationHeader {
+		http.SetCookie(w, &http.Cookie{
+			Name:     "receive-cookie-deprecation",
+			Value:    "1",
+			Secure:   true,
+			HttpOnly: true,
+			Path:     "/",
+			SameSite: http.SameSiteNoneMode,
+			// Partition: "",
+			Expires: time.Now().Add(time.Second * time.Duration(cookieDeprecationTTLSec)),
+		})
+	}
```

Comment:
Please refactor object creation to a helper method along the lines of `createReceiveCookieDeprecation` and use an the `timeutil` package to abstract the realtime clock for tests.

Reply:
done